topic-operator/src/test/java/io/strimzi/operator/topic/KafkaHandlerIT.java

```diff
@@ -218,6 +220,18 @@ public void shouldDescribeTopics() throws ExecutionException, InterruptedExcepti
         String t2Name = "should-describe-topics-2";
         createKafkaTopic(t2Name, Map.of());
 
+        // Wait until both topics actually exist
```

Comment:
Do the other tests in this class not have the same problem? Should the wait be inside the createKafkaTopic method?

Reply:
I tried to execute such tests 13 times in a row and didn't see any fail. But I think it would make more sense to move such logic inside `createKafkaTopic`?  But in that case we should re-name it accordingly? `createKafkaTopicWithWait` or smth similar?

Reply:
Yeah I think the name you suggested is good :)

Reply:
>I tried to execute such tests 13 times in a row and didn't see any fail. 

I wondered if that is partially due to the way the test is written. For example in the shouldDeleteTopics test we create three topics, fetch all the existing topics, delete the topics that are returned, then check those topics we sent for deletion were deleted. So if only 1 topic was created before the test continued I think the fetch would get back 1 topic, the call to delete the topics would include 1 topic and then the result would have 1 topic. If you added an assert to check that the Set of resultTopicNames is size 3 I wonder if you would see it fail sometimes.

Reply:
Looks like the issue is on TopicControllerIT, so why are we changing this class?

Reply:
> If you added an assert to check that the Set of resultTopicNames is size 3 I wonder if you would see it fail sometimes.

Yeah, I think that might be the problem.

Reply:
> Looks like the issue is on TopicControllerIT, so why are we changing this class?

While fixing the `TopicControllerIT` problem I also faced another one (i.e., inside `KafkaHandlerIT.java`), which I have fixed and tested accordingly.